packages/account-management/src/frontend/Registration/Registration.tsx

```diff
@@ -144,43 +165,34 @@ export const Registration: React.FC = () => {
               label="Last name"
               value={lastName}
               setValue={(value: string) => setLastName(value)}
+              isValid={lastName || lastName === ''}
+              showValidity={!!valid}
+              errorMessage={'Please enter your last name'}
             />
             <SpacingComponent />
-            <h1 className="font-wb font-size-4"> Login Details</h1>
+            <h1 className="font-wb font-size-4"> Login details</h1>
             <TextInput
               placeholder=""
               required={true}
               aria-label="Email Address"
               label="Email address"
+              isValid={!alreadyExists && emailValid}
+              showValidity={!!valid}
+              errorMessage={emailErrorMessage()}
               value={email}
               type="email"
               setValue={(value: string) => setEmail(value)}
             />
-            {alreadyExists ? (
-              <ErrorMessage>
-                This account already exists. You can try to <Link to='/'>login</Link>
-              </ErrorMessage>
-            ) : (
-              <></>
-            )}
-            {!emailValid ? <ErrorMessage>Please enter a valid email address.</ErrorMessage> : <></>}
             <SpacingComponent />
-            <PasswordInput value={pass} setValue={(value: string) => setPass(value)} pattern={passwordPolicy} />
-            {commonPassword ? (
-              <ErrorMessage>
-                The password you have entered has been flagged as a common password, or you have used your name in the
-                password. <br />
-                Please change your password and try again.
-              </ErrorMessage>
-            ) : (
-              <></>
-            )}
-            {!passQualifies ? (
-              <ErrorMessage>
-                The password you have entered does not meet the password policy. Please enter a password with at least 8
-                characters, a combination of upper and lowercase letters and at least one number.
-              </ErrorMessage>
-            ) : (
+            <PasswordInput
+              value={pass}
+              setValue={(value: string) => setPass(value)}
+              pattern={passwordPolicy}
+              isValid={!commonPassword && passQualifies}
+              showValidity={!!valid}
+              errorMessage={passwordErrorMessage()}
+            />
+            {!passQualifies ? null : (
```

Comment:
We could make this simpler:

```ts
{passQualifies && (
  <ul>
    // ...
  </ul>
)}
```

...but do we really only want to show the list of rules when the password qualifies?

Reply:
The password error message spells it out which is why I've hidden it?

Reply:
It's a UX call, but I think it's ok to do both. The user should see the rules before they start typing, and if there's an error it should explain why the password failed.

Reply:
I've taken out the condition here for now, as wireframes don't show either way. If there is a requirement to hide it when there is an error message shown, it's straightforward enough to add it in!